### PR TITLE
Updates to dev-python/construct

### PR DESCRIPTION
**Pull request by winterheart**
Update to latest 2.9.46, remove old. Dropped python 2.7 support and enabled tests.

### CODE REVIEW

**Comment by winterheart** [comment]
@juippis can you please review this PR?

File: dev-python/construct/construct-2.10.56.ebuild [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+DEPEND="test? (
+		dev-python/arrow[${PYTHON_USEDEP}]
+		>=dev-python/numpy-1.15.4[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/ruamel-yaml[${PYTHON_USEDEP}]
+	)"
+
+python_test() {
+	rm tests/test_benchmarks.py || die
```
Comment (mgorny) [inline dev-python/construct/construct-2.10.56.ebuild:27]: Move this to `src_prepare` and use `distutils_enable_tests`.

File: dev-python/construct/construct-2.10.56.ebuild [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+HOMEPAGE="https://construct.readthedocs.io/en/latest/ https://pypi.org/project/construct/"
+SRC_URI="https://github.com/construct/construct/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+DEPEND="test? (
```
Comment (mgorny) [inline dev-python/construct/construct-2.10.56.ebuild:19]: Move `test?` below to make indent more readable.

**Comment by winterheart** [comment]
@mgorny done.

**Comment by mgorny** [comment]
Please remove all the bits made redundant by d_e_t.

https://dev.gentoo.org/~mgorny/python-guide/distutils.html#using-distutils-enable-tests

**Comment by mgorny** [comment]
Looks good now, thanks.

**Pull request closed without merging**